### PR TITLE
Created checking function for xarray.DataArray and throw an error if not 

### LINKED ISSUE #14: Opaque error when a non-xarray object is passed to xhist
Currently when passing, say, an ndarray to `xhist`, the error that comes back is as follows:

```python
import numpy as np
from xhistogram.xarray import histogram as xhist

# Demo data
A = np.arange(100)
xhist(A, bins=[np.linspace(0, 100, 11)])
```

```python-traceback
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-75-0c14fe219de4> in <module>
----> 1 xhist(A, bins=[np.linspace(0, 100, 11)])

~/miniconda3/envs/analysis/lib/python3.6/site-packages/xhistogram/xarray.py in histogram(bins, dim, weights, density, block_size, bin_dim_suffix, bin_edge_suffix, *args)
     78     for a in args:
     79         # TODO: make this a more robust check
---> 80         assert a.name is not None, 'all arrays must have a name'
     81 
     82     # we drop coords to simplify alignment

AttributeError: 'numpy.ndarray' object has no attribute 'name'
```

This is minor, but it would be good to have a quick check that sees whether the object is an `xarray.DataArray` and if not, throw a clear error that that's the issue.

### PR DESCRIPTION
**Pull request by Badboy-16**
I had a go to fix #14 

In the `histogram()` (`xhist()`) function, I added some code to check if the object passed to the function is an instance of xarray.DataArray, and if not, throw a `TypeError`.

### CODE REVIEW

File: xhistogram/xarray.py [hunk 1]
```diff
@@ -106,8 +106,9 @@ def histogram(
     N_weights = 1 if weights is not None else 0
 
     for a in args:
-        # TODO: make this a more robust check
-        assert a.name is not None, "all arrays must have a name"
+        if not isinstance(a, xr.DataArray):
+            raise TypeError(f"xhist() accepts an xr.DataArray object but a " \
```
Comment (dougiesquire) [inline xhistogram/xarray.py:110]: Could you make this a bit more verbose:
f"xhistogram.xarray.histogram accepts only xarray.DataArray objects but a " +
f"{type(a).__name__} was provided"

**Review by dougiesquire — COMMENTED** [review]
Thanks @Badboy-16 and welcome to xhistogram!

A few very minor comments.

**Comment by dougiesquire** [comment]
Also, it looks as though your additions are failing the pre-commit test. You can run the tests locally to work out the issues by following the instructions [here](https://xhistogram.readthedocs.io/en/latest/contributing.html#preparing-pull-requests). Let me know if you have any difficulties.

**Comment by Badboy-16** [comment]
Thanks @dougiesquire 
I pushed a new commit regarding the above comments after the pre-commit test is successful locally. Please let me know if there are additional issues :)

**Comment by dougiesquire** [comment]
Great, pre-commit is passing now 👍. To keep the test coverage high, you could also add a test to xhistogram.tesy.test_xarray.py that checks your input test (sorry, I should have mentioned this sooner). Something like:
```python
def test_input_type_check():
    with pytest.raises(TypeError):
        # Try pass a numpy array to xhistogram
```

**Comment by Badboy-16** [comment]
It's ok. I just added a test in the latest commit. 😃

**Comment by dougiesquire** [comment]
Great - all checks passing now, so I'll merge. Thanks for your contribution @Badboy-16!

**Pull request merged**